tensorflow_datasets/core/load.py

```diff
@@ -332,6 +373,32 @@ def load(
   return ds
 
 
+def _get_builder_kwargs(path: str) -> Tuple[str, str, Optional[str], str]:
```

Comment:
It feels strange that the Builder require the parent dir.
Is it not possible to overwrite `def _build_data_dir(self, given_data_dir):` to return `(given_data_dir, given_data_dir)` instead ?
Similarly, instead of extracting the version and config from the path, we should try to load those info from the dataset_info.json

Reply:
Maybe at first you can try to fully overwrite __init__ and not call super() at all. Then we can see which modification would be required to DatasetBuilder.__init__ to support both regular and config-based loading.

Reply:
@Conchylicultor, I could not find any config related in `dataset_info.py`.
Am I missing out on something?